### PR TITLE
Add Stylelint as CSS/SCSS linter

### LINKED ISSUE #1036: [Suggestion] Linter for css/scss files
Currently, the website only has linter (eslint) only for js, ts, and tsx files.

CSS and SCSS files however, do not have a linter. I believe we can consider [stylelint](https://stylelint.io/) as the linter for CSS and SCSS files, as that would save the code reviewers' efforts to look through CSS styles. We can try our best to configure the linter to the style guide listed [here](https://www.thenewboston.com/style-guide/css)

We can incorporate the linter to the github action workflow done in https://github.com/thenewboston-developers/Website/pull/1033 as well.

If this is appropriate, do assign me for this task.

### PR DESCRIPTION
**Pull request by jamessspanggg**
Fixes #1036 

Installed the following packages: 
- stylelint
- stylelint-order (for the alphabetical order of properties)

Rules Applied: (All rules are taken from [here](https://stylelint.io/user-guide/rules/list)): 
- ["block-no-empty": true,](https://stylelint.io/user-guide/rules/block-no-empty)
- ["color-no-invalid-hex": true,](https://stylelint.io/user-guide/rules/color-no-invalid-hex)
- ["comment-no-empty": true,](https://stylelint.io/user-guide/rules/comment-no-empty)
-  ["declaration-block-single-line-max-declarations": 1,](https://stylelint.io/user-guide/rules/declaration-block-single-line-max-declarations)
- ["declaration-colon-space-after": "always",](https://stylelint.io/user-guide/rules/declaration-colon-space-after)
-  ["declaration-no-important": true,](https://stylelint.io/user-guide/rules/declaration-no-important)
-   ["function-comma-space-before": "never",](https://stylelint.io/user-guide/rules/function-comma-space-before)
- ["function-comma-space-after": "always",](https://stylelint.io/user-guide/rules/function-comma-space-after)
- ["indentation": 2,](https://stylelint.io/user-guide/rules/indentation)
- ["max-empty-lines": 2,](https://stylelint.io/user-guide/rules/max-empty-lines)
-  ["no-eol-whitespace": true,](https://stylelint.io/user-guide/rules/no-eol-whitespace)
-  ["no-extra-semicolons": true,](https://stylelint.io/user-guide/rules/no-extra-semicolons)
-  ["no-invalid-double-slash-comments": true,](https://stylelint.io/user-guide/rules/no-invalid-double-slash-comments)
-  "order/properties-alphabetical-order": true, (alphabetical order of css properties)
-  ["number-no-trailing-zeros": true,](https://stylelint.io/user-guide/rules/number-no-trailing-zeros)
-  ["property-case": "lower",](https://stylelint.io/user-guide/rules/property-case)
-  ["property-no-unknown": true,](https://stylelint.io/user-guide/rules/property-no-unknown)
-  ["selector-list-comma-newline-after": "always",](https://stylelint.io/user-guide/rules/selector-list-comma-newline-after)
-  ["selector-type-case": "lower",](https://stylelint.io/user-guide/rules/selector-type-case)
- ["string-no-newline": true,](https://stylelint.io/user-guide/rules/string-no-newline)
-  ["string-quotes": "single",](https://stylelint.io/user-guide/rules/string-quotes)
-  ["unit-no-unknown": true](https://stylelint.io/user-guide/rules/unit-no-unknown)

### CODE REVIEW

**Comment by jamessspanggg** [comment]
Before I continue to refactor the existing css codebase violations (most are alphabetical order of css properties), may I seek your opinions if the rules applied are sound? @Hristijan95 @thesanjeevsharma @lapstjup 

I added rules that are common and obvious to follow, but there are more rules in the [list](https://stylelint.io/user-guide/rules/list) that I did not add as I'm unsure.

**Comment by lapstjup** [comment]
@jamessspanggg These look good to me. If later we encounter a scenario which isn't covered, we will just add a rule. 
Also do you have to manually refactor the alphabetical order of css properties or is there a way you can use stylelint to fix them for you ? 
Also `"selector-type-case": "lower"` - Does this mean the class selector like `.NewsLetter` is also invalid ?

**Comment by jamessspanggg** [comment]
> @jamessspanggg These look good to me. If later we encounter a scenario which isn't covered, we will just add a rule.
> Also do you have to manually refactor the alphabetical order of css properties or is there a way you can use stylelint to fix them for you ?

Yes there is a way for stylelint to fix for you. Just need to provide the `--fix` argument while running stylelint. Speaking of this, should we convert `npm run prettier` to `npm run format` which executes both `prettier` (for js files) and `stylelint --fix` (for css files)?
 
> Also `"selector-type-case": "lower"` - Does this mean the class selector like `.NewsLetter` is also invalid ?

It will be valid, I ran it locally and it does not report the situation. I think the above only affects `li` or `ol`, i.e. without the `.` or `#` at the start. Should I remove this as it is slightly confusing?

**Comment by lapstjup** [comment]
@jamessspanggg Yeah you can remove that rule. 
Also, I think let's keep prettier separate for now. And a separate npm script for stylelint. 

For the stylelint - can you check whether inside the husky pre-commit hook of `lint-staged`, you can add something like :-
`stylelint --cache --fix` for css/scss regex after `prettier --write`,

If so, validate that the changed files with style linting errors when get commited are fixed by above.

File: package.json [hunk 1]
```diff
@@ -93,7 +94,8 @@
       "prettier --write"
     ],
     "{,!(node_modules)/**/}*.{css,scss}": [
-      "prettier --write"
+      "prettier --write",
+      "stylelint \"./src/**/*.{css,scss}\" --cache --fix"
```
Comment (lapstjup) [inline package.json:98]: Do we need the regex here again since its under the css, scss block ?

File: package.json [hunk 2]
```diff
@@ -41,6 +41,7 @@
     "eject": "react-scripts eject",
     "lint": "tsc --noEmit && eslint \"./src/**/*.{js,ts,tsx}\" && stylelint \"./src/**/*.{css,scss}\"",
     "prettier": "prettier --write \"{,!(node_modules)/**/}*.{ts,tsx,js,json,css,less,scss}\"",
+    "stylelint": "stylelint \"./src/**/*.{css,scss}\" --cache --fix",
```
Comment (lapstjup) [inline package.json:44]: For this I think let's keep it without cache.

File: package.json [hunk 3]
```diff
@@ -91,7 +94,8 @@
       "prettier --write"
     ],
     "{,!(node_modules)/**/}*.{css,scss}": [
-      "prettier --write"
+      "prettier --write",
+      "stylelint --cache --fix"
```
Comment (lapstjup) [inline package.json:98]: Were you able to test this ?

Comment (jamessspanggg) [reply package.json:98]: yes it works, i tried committing with violation and it auto resolves for me

**Review by lapstjup — APPROVED** [review]

**Pull request merged**